onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -54,6 +55,10 @@ TensorrtExecutionProvider::TensorrtExecutionProvider(const TensorrtExecutionProv
 
 TensorrtExecutionProvider::~TensorrtExecutionProvider() {}
 
+std::unique_ptr<onnxruntime::IDataTransfer> TensorrtExecutionProvider::GetDataTransfer() const {
+  return std::make_unique<onnxruntime::GPUDataTransfer>();
+}
+
```

Comment:
Given with this change there are multiple EPs registering the same data transfer type, should the data transfer manager be checking for that and only saving the first one? 

Or would it be better to use the old setup of registering the common data transfer types in inference session so we know they're only done once, and only using this override for custom/special EPs?

Reply:
I don't think we should assume the type of the data xfr object returned; we should only be concerned with the interface ptr returned by the GetDataTransfer() call. Granted that today since these live in the same codebase and are compiled together with ORT, we might be tempted to make such optimizations, however when we move to a true DLL based organization (where EPs can be developed, built and registered with ORT separately) we shouldn't be relying on this aspect. 
Besides I'm not sure if saving one less ptr helps ORT much since the # of providers used will be less than handful in all practical cases.

Reply:
The cost is in CanCopy calls to multiple instances of the same implementation. e.g. if both the CUDA EP and TensorRT register GPUDataTransfer, when looking for something for a CPU copy I could call CanCopy on the two GPUDataTransfer instances before getting to the CPUDataTransfer instance. 

I guess we can tweak it later to avoid this if necessary.

Reply:
Given the number of data xfr instances will most likely be less than 3 or 4 max, I guess we should measure if this is really a concern and can optimize later if needed.